src/test/java/org/springframework/data/couchbase/repository/query/PartTreeN1qBasedQueryTest.java

```diff
@@ -117,12 +126,129 @@ public void testGetCountExcludesDynamicSortClause() throws Exception {
 
 	}
 
+	@Test
```

Comment:
That looks like a lot of mocking going on here. Should this rather be an integration test?

To avoid the `ProjectionInformation` implementation, simply use `SpelAwareProxyProjectionFactory` and call `getProjectionInformation(…)` on it.